napari/layers/image/_slice.py

```diff
@@ -264,6 +264,27 @@ def _slice_indices_at_level(self, level: int) -> np.ndarray:
         indices[axes] = ds_indices
         return indices
 
+    def _get_slice_data(self):
+        slices = []
+        for dim_len, _point, low, high in zip(
+            self.data.shape,
+            self.data_slice.point,
+            self.data_slice.margin_left,
+            self.data_slice.margin_right,
+        ):
+            if np.isclose(high, low):
+                # assume slice thickness of 1 in data pixels
+                # (same as before thick slices were implemented)
+                high = low + 1
+
+            slice_ = slice(max(0, low), min(dim_len, high))
+            slices.append(slice_)
+
+        # TODO: for now just average is implemented over thick slice
+        return np.mean(
```

Comment:
Just a reminder, but this might explain the dtype issue with labels because slicing labels uses the image slice code.

Reply:
This code is quite different now: labels don't have this issue anymore cause they only project in `none` mode :)